### PR TITLE
Use Augur's `io.read_metadata`

### PR DESCRIPTION
**Pull request by huddlej**
## Description of proposed changes

Replaces all uses of `augur.utils.read_metadata` with the newer (and supported) `augur.io.read_metadata` function. This new function returns a pandas DataFrame instead of a dictionary, so many of the changes here update helper scripts to interact with DataFrames where possible.

## Related issue(s)

Related to https://github.com/nextstrain/augur/pull/934 and https://github.com/nextstrain/augur/issues/970

## Testing

 - Tested combined metadata with two input datasets
 - Confirmed that find clusters and recency annotation scripts produced identical output before and after changes

### CODE REVIEW

File: scripts/combine_metadata.py [hunk 1]
```diff
@@ -45,7 +44,12 @@ def parse_args():
     # READ IN METADATA FILES
     metadata = []
     for (origin, fname) in zip(args.origins, args.metadata):
-        data, columns = read_metadata(fname)
+        data = read_metadata(fname)
+        columns = ["strain"] + data.columns.tolist()
+        data = data.to_dict(orient="index")
+        for strain in data.keys():
+            data[strain]["strain"] = strain
```
Comment (victorlin) [inline scripts/combine_metadata.py:51]: This can also be accomplished using pandas functions:

```suggestion
        data = read_metadata(fname)
        data = data.rename_axis('strain').reset_index()
        data = data.to_dict(orient="index")
```

File: scripts/combine_metadata.py [hunk 2]
```diff
@@ -64,8 +68,8 @@ def parse_args():
     for strain in combined_data:
         for column in combined_columns:
             if column not in combined_data[strain]:
-                combined_data[strain][column] = EMPTY    
-    
+                combined_data[strain][column] = EMPTY
+
     for idx in range(1, len(metadata)):
         for strain, row in metadata[idx]['data'].items():
             if strain not in combined_data:
```
Comment (victorlin) [inline scripts/combine_metadata.py:75]: _(non-blocking)_

Some of this can probably be done more efficiently if data is kept as a DataFrame, but that should be a separate PR.

**Review by victorlin — CHANGES_REQUESTED** [review]

File: scripts/combine_metadata.py (hunk 2 above)
Comment (huddlej) [reply scripts/combine_metadata.py:75]: This script needs to be rewritten to handle larger metadata files at some point, anyway, so I figured pandas optimization could happen there.

File: scripts/combine_metadata.py (hunk 1 above)
Comment (huddlej) [reply scripts/combine_metadata.py:51]: I tried a similar approach, too, but it doesn't produce quite the same output, since resetting the index makes the index values integers again and then orienting the dict by index produces a dict indexed by those (meaningless) integers. Also, when you read in the file, the index is already named "strain".

The other approach that worked for me was (using data from the `nextstrain-ci` profile):

```python
data = read_metadata("results/europe/metadata_adjusted.tsv.xz")
data.insert(0, "strain", data.index.values)
columns = data.columns
data = data.to_dict(orient="index")
```

I can push this instead, if it seems clearer. It is probably marginally faster, although see other comment about the general need to optimize this script. :D

Comment (victorlin) [reply scripts/combine_metadata.py]: Ohh I see! That looks good.

**Review by victorlin — APPROVED** [review]

**Review by joverlee521 — APPROVED** [review]
